### ISSUE TITLE
`profile-desc` with non-ASCII characters cause `--profile=help` and `--show-profile=xxx`command to block

### ISSUE THREAD

**Issue by verygoodlee** [mpv-player/mpv#13253]
### Important Information
Provide following Information:
- mpv version
```
mpv v0.37.0-144-gd8c2e33a Copyright © 2000-2023 mpv/MPlayer/mplayer2 projects
 built on Jan  5 2024 00:10:14
libplacebo version: v6.338.0-62-g52314e0-dirty
FFmpeg version: N-113214-g8b76bae89
FFmpeg library versions:
   libavutil       58.36.101
   libavcodec      60.37.100
   libavformat     60.20.100
   libswscale      7.6.100
   libavfilter     9.16.100
   libswresample   4.13.100
```
- Windows Version
  Win10 22H2
- Source of the mpv binary
  https://github.com/shinchiro/mpv-winbuild-cmake/releases/tag/20240105
### Reproduction steps
1. `profile-desc` with Chinese character
```
# mpv.conf
[non-ASCII-test]
profile-desc=去色带
deband=yes
```
2. Run `mpv --profile=help` or `mpv --show-profile=non-ASCII-test` in Terminal 

### Expected behavior
In older version https://github.com/shinchiro/mpv-winbuild-cmake/releases/tag/20231214
![image](https://github.com/mpv-player/mpv/assets/45035465/64a55e1f-64e7-4262-84ca-27e68fb653da)

### Actual behavior
The Terminal is blocked and can only be interrupted by `Ctrl+C`
![image](https://github.com/mpv-player/mpv/assets/45035465/a2266be0-ef9e-4c4f-9c0d-6803d631eab0)
High CPU usage during blocking
![image](https://github.com/mpv-player/mpv/assets/45035465/4826a5ab-238b-4737-91a0-015ce760e95b)

**Comment by verygoodlee** [comment]
Changing the `mpv.conf` file encoding to utf-8 solves this problem, in old versions utf-8 encoding is not required.

**Comment by guidocella** [comment]
Should be fixed by 485221ba22

**Comment by verygoodlee** [comment]
> Should be fixed by [485221b](https://github.com/mpv-player/mpv/commit/485221ba22674045d1af822b08fa193862dd835b)

Yes, test passed in https://github.com/shinchiro/mpv-winbuild-cmake/releases/tag/20240110 .
